### PR TITLE
Messaging gcp pubsub

### LINKED ISSUE #369: Support for Google Cloud Platform Pub/Sub transport
Hello,
I was wondering if you are willing to review/accept a new transport for GCP Pub/Sub? I am currently working from my fork here: https://github.com/newportd/smallrye-reactive-messaging/tree/messaging-gcp-pubsub

### PR DESCRIPTION
**Pull request by newportd**
This pull request adds a new transport for Google Cloud Platform Pub/Sub.

Resolves #369

### CODE REVIEW

File: smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            .ignore();
+    }
+
+    private void createTopic(final PubSubConfig config) {
+        final TopicAdminClient topicAdminClient = pubSubManager.topicAdminClient(config);
+
+        final ProjectTopicName topicName = ProjectTopicName.of(config.getProjectId(), config.getTopic());
+
+        try {
+            topicAdminClient.getTopic(topicName);
```
[the remark this answers is not in the document]
Comment (newportd) [reply smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java]: Yes, they are. I will refactor this and the subscription calls to happen asynchronously.

File: smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+        final Publisher publisher = pubSubManager.publisher(pubSubConfig);
+
+        return ReactiveStreams.<Message<?>>builder()
+            .flatMapCompletionStage(message -> {
+                final PubsubMessage pubSubMessage = buildMessage(message);
+
+                final Future<String> future = publisher.publish(pubSubMessage);
+
+                return CompletableFuture.supplyAsync(await(future));
```
[the remark this answers is not in the document]
Comment (newportd) [reply smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java]: It is not. I will refactor this.

[the remark this answers is not in the document]
Comment (newportd) [reply smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java]: I've pushed this refactor. Please take a look and let me know your thoughts.

File: smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (newportd) [reply smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java]: I've pushed this refactor as well.

**Comment by cescoffier** [comment]
Look great! Let me merge it like it is. If there are some improvements, just create another PR.

**Pull request merged**